src/pages/index.js

```diff
@@ -62,36 +104,160 @@ function handleImageClick(cardData) {
 
 //edit profile
 function handleProfileSubmit(data) {
-  userInfo.setUserInfo({name: data.name, job: data.description});
-editPopup.close(); }
+  renderLoading(true, editPopup);
+  api.updateUserInfo({
+    name: data.name,
+    about: data.about
+  })
+    .then(updatedUserData => {
+      user.setUserInfo({
+        name: updatedUserData.name, 
+        about: updatedUserData.about || updatedUserData.dascription || updatedUserData.bio
+      });
+    })
+    .catch(err => console.error(`Error updating profile: ${err}`))
+    .finally(() =>{
+      renderLoading(false, editPopup);
+      editPopup.close();
```

Comment:
`      editPopup.close();
`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well

Reply:
* You haven't corrected my remark https://github.com/Schou10/se_project_aroundtheus/pull/1#discussion_r1747450964  You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well


**Please, correct it everywhere in the code**